src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -570,25 +571,22 @@ public void filterForCourseDashboard(List<StudentParticipation> participations,
             }
         }
 
-        // add results to participation
+        // add relevant submission with its result to participation
```

Comment:
I don't like the wording "relevant" in general in this class. Please document what relevant means, when a submission is relevant and how you find an "appropriate" submission (again, "appropriate" should also be extensively documented)

Reply:
I renamed `findAppropriateSubmission()` and extended the documentation for it. Since I didn't create the method `findRelevantParticipation()` I haven't renamed it and I only improved the inline documentation.